### DISCUSSION TITLE
Question: API to access SMTP Header Message ID?

### DISCUSSION THREAD

**Discussion by rajendraApollo** [InboxSDK/InboxSDK#1114] [Ideas]
Hi there,
We are currently using `"@inboxsdk/core": "1.1.6"`.
And we have a use case whereby we need access to `SMTP Header Message ID`
<img width="1130" alt="Screenshot 2023-08-30 at 11 11 16 AM" src="https://github.com/InboxSDK/InboxSDK/assets/99239793/b5c37bd3-6a8d-426e-8ecb-02ff3a0dec19">

Is there any exiting API that we can use to get those header details for an email.
If not do you see this as something that will be added in the near future? (If yes, approx timeline 😅 )

Also if you know any possible workaround, that would really help.

Thanks in advance.

**Comment by wegry** [comment]
PRs welcome.

**Comment by aleemstreak** [comment]
@rajendraApollo the InboxSDK focuses mainly on data that the user can see when using Gmail. To get headers, attachments, etc you should look into the Gmail API as that is much more robust way of getting gmail data. 

The RFC message ID that you're looking for may be an exception (the SDK does try to find as many IDs as possible so that developers can tie what the user is looking at to their backend data sets that come from things like the Gmail API). 

The SDK may already be parsing this but not exposing it, PRs def welcome.
